haystack/components/caching/cache_checker.py

```diff
@@ -82,7 +82,7 @@ def from_dict(cls, data: Dict[str, Any]) -> "CacheChecker":
             doc_store_class = import_class_by_name(doc_store_data["type"])
         except ImportError as e:
             raise DeserializationError(f"Class '{doc_store_data['type']}' not correctly imported") from e
-        data["init_parameters"]["document_store"] = default_from_dict(doc_store_class, doc_store_data)
+        data["init_parameters"]["document_store"] = doc_store_class.from_dict(doc_store_data)  # type: ignore[attr-defined]
```

Comment:
mypy says:
> "type[object]" has no attribute "from_dict"

there are several ways to fix this.
I would have liked to check if the `doc_store_class` respects the `DocumentStore` protocol, but our protocol is not `@runtime_checkable`...

Reply:
You could also check if the `doc_store_class` has the `from_dict` function, otherwise call the `default_to_dict` on the `doc_store_class` or fallback on exception to `default_to_dict`.